sycl/doc/extensions/SPIRV/SPV_INTEL_long_constant_composite.asciidoc

```diff
@@ -0,0 +1,235 @@
+SPV_INTEL_long_constant_composite
+=================================
+
+Name Strings
+------------
+
+SPV_INTEL_long_constant_composite
+
+Contact
+-------
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+Contributors
+------------
+
+- Mariya Podchishchaeva, Intel
+- Alexey Sotkin, Intel
+- Ben Ashbaugh, Intel
+- Alexey Sachkov, Intel
+
+Notice
+------
+
+Copyright (c) 2020 Intel Corporation. All rights reserved.
+
+Status
+------
+
+Draft.
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2020-12-07
+| Revision           | 6
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.5 Revision 4, Unified
+
+This extension requires SPIR-V 1.0.
+
+Overview
+--------
+
+This extension adds new capability and instructions to allow to represent
+composite constants with number of Constituents greater than the maximum
+possible WordCount.
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_long_constant_composite"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5
+------------------------------------------------------
+Capabilities
+~~~~~~~~~~~~
+
+Modify Section 3.31, "Capability", adding these rows to the Capability table:
+
+--
+[options="header"]
+|====
+2+^| Capability ^| Depends On
+| 6089 | *CapabilityLongConstantCompositeINTEL* +
+Allow to use *OpTypeStructContinuedINTEL*, *OpConstantCompositeContinuedINTEL*
+and *OpSpecConstantCompositeContinuedINTEL* instructions |
+|====
+--
+
+Instructions
+~~~~~~~~~~~~
+In section 3.37.6. Type-Declaration Instructions add the new instruction
+[cols="3", width="100%"]
+|=====
+2+^|*OpTypeStructContinuedINTEL* +
+
+Continue specifying an *OpTypeStruct* with number of _Member types_
+greater than the maximum possible WordCount.
+
+The previous instruction must be an *OpTypeStruct* or an
+*OpTypeStructContinuedINTEL* instruction.
+
+_Member types_ follow the same rules as defined for _Member types_ of
+*OpTypeStruct*.
+
+| Capability:
+*CapabilityLongConstantCompositeINTEL*
+
+| 2 + variable | 6090 | _<id>, <id>, ... Member N type, member N + 1 type_|
+|=====
+
+Modify the description of *OpTypeStruct* instruction, adding the
+following sentence to the end:
+In case if it is not possible to specify all the _member types_ of the structure
+by one *OpTypeStruct* instruction, i.e. if number of members of the
+_Result type_ is greater than the maximum possible WordCount, the remaining
```

Comment:
Please mention, that _Result type_ is a result type of OpConstantComposite instruction, otherwise it looks like that OpTypeStructContinuedINTEL has a result.

Also when the file gets viewed some table rows are missing.

Reply:
Could you please elaborate what is the problem with table rows? I probably don't see where it is.

Reply:
Nevermind, I missed to where "Modify the description of OpTypeStruct instruction..." belongs.